src/catalog/src/durable/impls/persist/metrics.rs

```diff
@@ -0,0 +1,49 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+//! Prometheus monitoring metrics.
+
+use mz_ore::metric;
+use mz_ore::metrics::{IntCounter, MetricsRegistry};
+use mz_ore::stats::histogram_seconds_buckets;
+use prometheus::{Histogram, IntCounterVec};
+
+#[derive(Debug, Clone)]
+pub struct Metrics {
+    pub transactions: IntCounter,
+    pub transaction_commit_errors: IntCounterVec,
+    pub transaction_commit_latency_duration_seconds: Histogram,
```

Comment:
Histograms are quite a bit more expensive than a counter, specifically they are equal to `N=2` counters where N is the number of buckets in the histogram. Plus, you can get pretty far by just using carefully selected normal counters. So, my rule of thumb is to never use a histogram until there's been a debugging incident that couldn't be diagnosed without the histogram. For reference, all of persist has 6 histograms (and I think one of them doesn't actually need to be a histogram)

Concretely, I think here you should start by replacing each of these histograms with one counter for the total number of seconds spend and one for the number of calls. That's enough to get you (1) number of calls (2) total time spent and (3) average time spent, which might very well be all you ever need

Reply:
Done, let me know if the counter approach I used looks good.